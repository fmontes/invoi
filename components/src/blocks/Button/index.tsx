import * as React from 'react';
import styled from 'styled-components';
import theme, { ThemeModel } from '../../theme';

type ButtonProps = {
    naked?: boolean;
    disabled?: boolean;
    secondary?: boolean;
    tiny?: boolean;
    theme: ThemeModel;
    children: React.ReactNode;
};

const ButtonStyled = styled.button<ButtonProps>`
    background-color: ${({ theme, secondary, naked }: ButtonProps) =>
        naked ? 'transparent' : secondary ? theme.colors.gray : theme.colors.primary};
    border-radius: ${({ theme, tiny }: ButtonProps) => theme.base.radius[tiny ? 'tiny' : 'normal']};
    border: 0;
    color: ${({ theme, naked }: ButtonProps) =>
        naked ? theme.colors.textBody : theme.colors.white};
    font-family: ${({ theme }: ButtonProps) => theme.font.family};
    font-size: ${({ theme, tiny }: ButtonProps) => theme.font.size[tiny ? 'tiny' : 'body']};
    height: ${({ theme, tiny }: ButtonProps) =>
        theme.components.buttons[tiny ? 'tiny' : 'normal'].height};
    padding: 0 ${({ theme, tiny }: ButtonProps) => theme.spaces[tiny ? 'one' : 'two']};
    text-transform: ${({ tiny }: ButtonProps) => (tiny ? 'uppercase' : 'normal')};

    &:disabled {
        background-color: ${({ theme }: ButtonProps) => theme.colors.disabled};
    }
`;

ButtonStyled.defaultProps = {
    theme: theme
};

export function Button(props: ButtonProps): JSX.Element {
    return <ButtonStyled {...props}>{props.children}</ButtonStyled>;
}
