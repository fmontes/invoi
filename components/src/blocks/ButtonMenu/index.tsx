import * as React from 'react';
import styled from 'styled-components';
import theme, { ThemeModel } from '../../theme/index';
import { Add } from '@material-ui/icons';

type ButtonMenuProps = {
    disabled?: boolean;
    theme: ThemeModel;
};

const ButtonMenuStyled = styled.button<ButtonMenuProps>`
    align-items: center;
    background-color: ${({ theme }: ButtonMenuProps) => theme.colors.primary};
    border-radius: 50%;
    border: 0;
    color: ${({ theme }: ButtonMenuProps) => theme.colors.white};
    display: inline-flex;
    height: 64px;
    justify-content: center;
    width: 64px;

    &:disabled {
        background-color: ${({ theme }: ButtonMenuProps) => theme.colors.disabled};
    }
`;

ButtonMenuStyled.defaultProps = {
    theme: theme
};

export function ButtonMenu(props: ButtonMenuProps): JSX.Element {
    return (
        <ButtonMenuStyled {...props}>
            <Add style={{ fontSize: 32 }} />
        </ButtonMenuStyled>
    );
}
