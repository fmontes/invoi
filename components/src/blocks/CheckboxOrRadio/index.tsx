import * as React from 'react';
import styled from 'styled-components';
import theme, { ThemeModel } from '../../theme/index';
import { transition, fieldBoxes } from '../../theme/utils';

export interface CheckboxOrRadioProps {
    checked: boolean;
    disabled: boolean;
    error: boolean;
    id?: string;
    name?: string;
    theme: ThemeModel;
    type: string;
};

const CheckboxOrRadioStyled = styled.input`
    height: 20px;
    margin: 0;
    padding: 0;
    position: absolute;
    width: 20px;
    z-index: 1;
    appearance: none;

    & + span {
        ${({ theme, error }: CheckboxOrRadioProps) => fieldBoxes.normal(theme, error)};
        ${({ type }: CheckboxOrRadioProps) => type === 'radio' ? 'border-radius: 50%' : null };
        display: block;
        height: 20px;
        width: 20px;
        margin-right: ${({ theme }: CheckboxOrRadioProps) => theme.spaces.one};
        position: relative;

        &:after {
            ${({ type }: CheckboxOrRadioProps) => type === 'radio' ? 'border-radius: 50%' : null };
            content: '';
            height: 10px;
            left: 50%;
            position: absolute;
            top: 50%;
            transform: scale(0) translate(-50%, -50%);
            transition: ${({ theme }: CheckboxOrRadioProps) =>
                transition(
                    theme,
                    ['transform', 'background-color'],
                    'cubic-bezier(.74,-0.29,.33,1.63)'
                )};
            width: 10px;
            transform-origin: 0 0;
        }
    }

    &:checked + span {
        border-color: ${({ theme }: CheckboxOrRadioProps) => theme.colors.primary};

        &:after {
            transform: scale(1) translate(-50%, -50%);
            background-color: ${({ theme }: CheckboxOrRadioProps) => theme.colors.primary};
        }
    }

    &:not([disabled]) {
        cursor: pointer;
        &:focus + span,
        &:hover + span {
            ${({ theme }: CheckboxOrRadioProps) => fieldBoxes.focus(theme)};
        }

        &:active + span {
            ${({ theme }: CheckboxOrRadioProps) => fieldBoxes.active(theme)};
        }
    }

    &:disabled + span {
        ${({ theme }: CheckboxOrRadioProps) => fieldBoxes.disabled(theme)};

        &:after {
            background-color: ${({ theme }: CheckboxOrRadioProps) => theme.colors.disabled};
        }
    }
`;

CheckboxOrRadioStyled.defaultProps = {
    theme: theme,
    error: false
};

export function CheckboxOrRadio(props: CheckboxOrRadioProps): JSX.Element {
    return (
        <>
            <CheckboxOrRadioStyled {...props} /> <span />
        </>
    );
}
