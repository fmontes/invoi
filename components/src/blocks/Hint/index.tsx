import * as React from 'react';
import styled from 'styled-components';
import theme, { ThemeModel } from '../../theme';

interface HintProps {
    children: React.ReactNode;
    disabled?: boolean;
    error: boolean;
    theme: ThemeModel;
};

const HintStyled = styled.span<HintProps>`
    color: ${({ theme, disabled, error }: HintProps) => disabled ? theme.colors.disabled : error ? theme.colors.error : ''};
    font-family: ${({ theme }: HintProps) => theme.font.family};
    font-size: ${({ theme }: HintProps) => theme.font.size.tiny};
`;

HintStyled.defaultProps = {
    disabled: false,
    error: false,
    theme: theme
};

const Hint: React.FC<HintProps> = (props: HintProps) => {
    return <HintStyled {...props}>{props.children}</HintStyled>;
}

export default Hint;
