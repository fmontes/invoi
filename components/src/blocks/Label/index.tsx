import * as React from 'react';
import styled from 'styled-components';
import theme, { ThemeModel } from '../../theme';

type LabelProps = {
    children: React.ReactNode;
    disabled?: boolean;
    error?: boolean;
    htmlFor?: string;
    theme: ThemeModel;
};

const LabelStyled = styled.label<LabelProps>`
    color: ${({ theme, disabled, error }: LabelProps) => disabled ? theme.colors.disabled : error ? theme.colors.error : ''};
    font-family: ${({ theme }: LabelProps) => theme.font.family};
    font-size: ${({ theme }: LabelProps) => theme.font.size.body};
    line-height: 1.5;
`;

LabelStyled.defaultProps = {
    disabled: false,
    error: false,
    theme: theme
};

const Label: React.FC<LabelProps> = (props: LabelProps) => {
    return <LabelStyled {...props}>{props.children}</LabelStyled>;
};

export default Label;
