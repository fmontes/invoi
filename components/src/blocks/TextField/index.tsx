import * as React from 'react';
import styled from 'styled-components';
import theme, { rgba, transition, ThemeModel } from '../../theme';

type TextFieldProps = {
    theme: ThemeModel;
    type?: string;
    id?: string;
    error: boolean;
};

const TextFieldStyled = styled.input<TextFieldProps>`
    border: solid 2px
        ${({ theme, error }: TextFieldProps) => (error ? theme.colors.error : theme.colors.neutral)};
    border-radius: 2px;
    box-sizing: border-box;
    font-family: ${({ theme }: TextFieldProps) => theme.font.family};
    font-size: ${({ theme }: TextFieldProps) => theme.font.size.body};
    height: ${({ theme }: TextFieldProps) => theme.components.buttons.normal.height};
    padding: 0 ${({ theme }: TextFieldProps) => theme.spaces.one};
    transition: ${transition(['border', 'box-shadow'])};

    &:focus {
        border-color: ${({ theme }: TextFieldProps) => theme.colors.primary};
        box-shadow: 0 0 20px 0 ${rgba('primary', '0.2')};
    }

    &:active {
        border-color: ${({ theme }: TextFieldProps) => theme.colors.primary};
        box-shadow: 0 0 0 4px ${rgba('primary', '0.1')};
    }

    &:disabled {
        background: none;
        border-style: dotted;
        border-color: ${({ theme }: TextFieldProps) => theme.colors.disabled};
    }
`;

TextFieldStyled.defaultProps = {
    theme: theme,
    type: 'text',
    error: false
};

export function TextField(props: TextFieldProps): JSX.Element {
    return <TextFieldStyled {...props} />;
}
