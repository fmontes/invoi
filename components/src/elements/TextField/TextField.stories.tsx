import React from 'react';
import { storiesOf } from '@storybook/react';
import { TextField } from './index';

storiesOf('TextField Normal', module)
    .add('normal', () => <TextField type="text" />)
