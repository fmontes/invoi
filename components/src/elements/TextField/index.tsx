import * as React from 'react';
import styled from 'styled-components';
import theme from '../../theme';

type TextFieldProps = {
    theme?: any;
    type: string;
};

const TextFieldStyled = styled.input<TextFieldProps>`
    border: solid 2px ${({ theme }) => theme.colors.neutral};
    box-sizing: border-box;
    font-family: ${({ theme }) => theme.font.family};
    font-size: ${({ theme }) => theme.font.size.body};
    height: ${({ theme }) => theme.components.buttons.normal.height};
    padding: 0 ${({ theme }) => theme.spaces.one};
`;

TextFieldStyled.defaultProps = {
    theme: theme,
    type: 'text'
};

export function TextField(props: TextFieldProps): JSX.Element {
    return <TextFieldStyled {...props} />;
}
