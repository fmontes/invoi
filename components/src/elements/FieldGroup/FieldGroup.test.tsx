import * as React from 'react';
import { FieldGroup } from './index';
import { shallow } from 'enzyme';
import { TextField } from '../../blocks/TextField';

test('FieldGroup should render', () => {
    const component = shallow(
        <FieldGroup id="email" label="Hello World">
            <TextField id="email" type="email" />
        </FieldGroup>
    );
    expect(component).toMatchSnapshot();
});

test('FieldGroup should render with hint', () => {
    const component = shallow(
        <FieldGroup hint="Hello Im the hint" id="email" label="Hello World">
            <TextField id="email" type="email" />
        </FieldGroup>
    );
    expect(component).toMatchSnapshot();
});
