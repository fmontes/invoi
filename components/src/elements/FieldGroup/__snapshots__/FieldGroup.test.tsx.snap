// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldGroup should render 1`] = `
<styled.div
  disabled={false}
  error={false}
  id="email"
  label="Hello World"
  theme={
    Object {
      "animation": Object {
        "speed": "100ms",
      },
      "base": Object {
        "radius": Object {
          "normal": "0.5rem",
          "tiny": "0.25rem",
        },
      },
      "colors": Object {
        "background": "#FDFCF9",
        "black": "#202641",
        "disabled": "#D7D0CB",
        "error": "#832400",
        "gray": "#5B6078",
        "neutral": "#CBC1BB",
        "primary": "#3951C5",
        "secondary": "#F0A78B",
        "success": "#3951C5",
        "textBody": "#202641",
        "textSub": "#5B6078",
        "white": "#fff",
      },
      "components": Object {
        "buttons": Object {
          "normal": Object {
            "height": "2.75rem",
          },
          "tiny": Object {
            "height": "1.5rem",
          },
        },
      },
      "font": Object {
        "family": "'Work Sans', sans-serif;",
        "size": Object {
          "body": "1rem",
          "h1": "2rem",
          "h2": "1.5rem",
          "hint": "0.8125rem",
          "tiny": "0.6875rem",
        },
      },
      "spaces": Object {
        "half": "0.25rem",
        "one": "0.5rem",
        "three": "2rem",
        "two": "1rem",
      },
    }
  }
>
  <Label
    className="label"
    id="email"
    label="Hello World"
  >
    Hello World
  </Label>
  <TextField
    id="email"
    type="email"
  />
</styled.div>
`;

exports[`FieldGroup should render with hint 1`] = `
<styled.div
  disabled={false}
  error={false}
  hint="Hello Im the hint"
  id="email"
  label="Hello World"
  theme={
    Object {
      "animation": Object {
        "speed": "100ms",
      },
      "base": Object {
        "radius": Object {
          "normal": "0.5rem",
          "tiny": "0.25rem",
        },
      },
      "colors": Object {
        "background": "#FDFCF9",
        "black": "#202641",
        "disabled": "#D7D0CB",
        "error": "#832400",
        "gray": "#5B6078",
        "neutral": "#CBC1BB",
        "primary": "#3951C5",
        "secondary": "#F0A78B",
        "success": "#3951C5",
        "textBody": "#202641",
        "textSub": "#5B6078",
        "white": "#fff",
      },
      "components": Object {
        "buttons": Object {
          "normal": Object {
            "height": "2.75rem",
          },
          "tiny": Object {
            "height": "1.5rem",
          },
        },
      },
      "font": Object {
        "family": "'Work Sans', sans-serif;",
        "size": Object {
          "body": "1rem",
          "h1": "2rem",
          "h2": "1.5rem",
          "hint": "0.8125rem",
          "tiny": "0.6875rem",
        },
      },
      "spaces": Object {
        "half": "0.25rem",
        "one": "0.5rem",
        "three": "2rem",
        "two": "1rem",
      },
    }
  }
>
  <Label
    className="label"
    hint="Hello Im the hint"
    id="email"
    label="Hello World"
  >
    Hello World
  </Label>
  <TextField
    id="email"
    type="email"
  />
  <Hint
    className="hint"
    hint="Hello Im the hint"
    id="email"
    label="Hello World"
  >
    Hello Im the hint
  </Hint>
</styled.div>
`;
